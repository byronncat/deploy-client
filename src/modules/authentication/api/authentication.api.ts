import axios from 'axios';
import { uri } from '@global';

import type { AxiosResponse } from 'axios';
import type { API } from '@global';
import type { LoginFormData, RegisterFormData } from '../types';

export async function login(data: LoginFormData): Promise<API> {
  return await axios
    .post(uri.getHostingServer('login'), data)
    .then((res: AxiosResponse) => res.data)
    .catch((error) => error.response.data);
}

export async function register(data: RegisterFormData): Promise<API> {
  return await axios
    .post(uri.getHostingServer('register'), data)
    .then((res: AxiosResponse) => res.data)
    .catch((error) => error.response.data);
}

export async function logout(): Promise<API> {
  return await axios
    .delete(uri.getHostingServer('logout'))
    .then((res: AxiosResponse) => res.data)
    .catch((error) => error.response.data);
}

export async function authenticate(): Promise<API> {
  return await axios
    .get(uri.getHostingServer('authenticate'))
    .then((res: AxiosResponse) => res.data)
    .catch((error) => error.response.data);
}
