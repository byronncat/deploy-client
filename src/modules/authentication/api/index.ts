export * as authenticationApi from './authentication.api';
