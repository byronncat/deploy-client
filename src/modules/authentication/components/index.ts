export { default as Divider } from './Divider.component';
export { default as Form } from './form/Form.component';
export { default as NavigationText } from './NavigationText.component';
