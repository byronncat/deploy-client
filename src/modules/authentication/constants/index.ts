export * from './defaultValues.constant';
export * from './formFields.constant';
