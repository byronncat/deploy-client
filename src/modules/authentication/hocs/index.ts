export { default as Authentication } from './Authentication.hoc';
export { default as Landing } from './Landing.hoc';
export * from './Landing.hoc';
