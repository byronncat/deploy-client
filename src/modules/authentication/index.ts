export * from './api';
export * from './components';
export * from './constants';
export * from './hocs';
export * from './layouts';
export * from './pages';
export * from './providers';
export * from './types/index.d';
