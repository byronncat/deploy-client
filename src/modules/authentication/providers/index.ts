export {
  default as ClassNameProvider,
  ClassNameContext,
} from './Classname.context';

export {
  default as AuthenticationProvider,
  AuthContext,
} from './Authentication.provider';
