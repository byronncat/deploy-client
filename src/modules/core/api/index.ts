export * as postApi from './post.api';
export * as profileApi from './profile.api';
