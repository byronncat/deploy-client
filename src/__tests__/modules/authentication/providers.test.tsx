import {
  useClassNameContext,
  AuthenticationProvider,
  useAuthContext,
  ClassNameProvider,
} from '@authentication';
import { render, renderHook, waitFor } from '@testing-library/react';
import { ReactProps } from '@global';
import { authenticationApi } from '@/modules/authentication/api';
import { useEffect } from 'react';

jest.mock('@/modules/authentication/api');
describe('provider', () => {
  describe('className', () => {
    it('should return className object', async () => {
      const wrapper = ({ children }: ReactProps) => (
        <ClassNameProvider>{children}</ClassNameProvider>
      );
      const { result } = await renderHook(() => useClassNameContext(), {
        wrapper,
      });

      expect(result.current.className).toMatchSnapshot();
    });
  });

  // The issue might be related to the scope of the mock.
  describe('authentication', () => {
    beforeEach(() => {
      (authenticationApi.login as jest.Mock).mockResolvedValue({
        success: true,
        message: '',
      });
    });

    it('should return isAuthenticated object', async () => {
      let result: boolean = false;
      const Test = () => {
        const { isLoggedIn: isAuthenticated } = useAuthContext();

        useEffect(() => {
          result = isAuthenticated;
        }, [isAuthenticated]);
        return (
          <div data-testid="authentication">{isAuthenticated.toString()}</div>
        );
      };
      render(
        <AuthenticationProvider>
          <Test />
        </AuthenticationProvider>,
      );

      await waitFor(() => {
        expect(result).toBe(true);
      });
    });
  });
});
