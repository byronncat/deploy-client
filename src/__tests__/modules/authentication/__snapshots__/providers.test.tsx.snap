// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`provider className should return className object 1`] = `
Object {
  "form": "form-background max-w-full w-100 p-10 pb-8 rounded-lg shadow-xl",
  "formErrorMessage": "mt-1 mb-2 font-medium text-neon-red",
  "formErrorMessageAnimation": "flicker-once",
  "formField": "relative my-3",
  "formInput": "block w-full px-3 pt-5 pb-2 rounded bg-gray-50 dark:bg-gray-700 border-0 border-b-2 appearance-none border-gray-600 text-white text-sm transition duration-300 focus:border-slate-400 focus:outline-none focus:ring-0 peer",
  "formLabel": "absolute start-3 top-4 z-10 text-gray-400 text-sm duration-300 -translate-y-4 scale-75 origin-[0] peer-focus:text-white peer-placeholder-shown:scale-100 peer-placeholder-shown:translate-y-0 peer-focus:scale-75 peer-focus:-translate-y-4",
}
`;
